numpy/lib/function_base.py

```diff
@@ -3482,6 +3483,11 @@ def median(a, axis=None, out=None, overwrite_input=False, keepdims=False):
     >>> assert not np.all(a==b)
 
     """
+    if isinstance(a, np.ma.MaskedArray):
+        a = a.data
+        warnings.warn("Warning: 'median' will ignore the 'mask' "
+                      "of the MaskedArray.", stacklevel=1)
```

Comment:
Should be `stacklevel=2`, and should not have a `Warning:` prefix, as python will add that automatically. Same throughout.

Reply:
The problem is that those methods are not implemented in ``ndarray`` and it's impossible to overwrite them in ``MaskedArray`` and add warning. Another approach is to just leave a comment in documentation about undefined behavior of those methods on MaskedArrays.
It might be a good idea to create another issue on implementation of ``partition`` and ``argpartition`` for MaskedArrays, which would solve that problem too. But for now, warning is the simplest solution.